core/src/main/java/org/elasticsearch/common/settings/Setting.java

```diff
@@ -61,32 +65,99 @@
  * public enum Color {
  *     RED, GREEN, BLUE;
  * }
- * public static final Setting<Color> MY_BOOLEAN = new Setting<>("my.color.setting", Color.RED.toString(), Color::valueOf, false, Scope.CLUSTER);
+ * public static final Setting<Color> MY_BOOLEAN =
+ *     new Setting<>("my.color.setting", Color.RED.toString(), Color::valueOf, SettingsProperty.ClusterScope);
  * }
  * </pre>
  */
 public class Setting<T> extends ToXContentToBytes {
+
+    public enum SettingsProperty {
+        /**
+         * should be filtered in some api (mask password/credentials)
+         */
+        Filtered,
+
+        /**
+         * iff this setting can be dynamically updateable
+         */
+        Dynamic,
+
+        /**
+         * mark this setting as deprecated
+         */
+        Deprecated,
+
+        /**
+         * Cluster scope.
+         * @See IndexScope
+         * @See NodeScope
+         */
+        ClusterScope,
```

Comment:
Done. I removed `ClusterScope` and replaced it by `NodeScope`. Also renamed `ClusterSettings` to `NodeSettings`.

I believe that at some point, we can have some node settings that could be dynamic. For example, let say that I'm using s3 repository and that I want to update my credentials.
The only way to do that now is to change `elasticsearch.yml` and restart all the nodes.

May be we could think of a more elegant feature in the future. So may be we will need to introduce again the `ClusterScope` notion?